activerecord/lib/active_record/schema.rb

```diff
@@ -54,8 +54,8 @@ def define(info = {}, &block)
       def define(info, &block) # :nodoc:
         instance_eval(&block)
 
+        connection.schema_migration.create_table
```

Comment:
Moved this outside of the condition so that this table is always created.
Since all Active Record test cases inherit from `ActiveRecord::TestCase`, we always have this table available in tests via https://github.com/rails/rails/blob/13783f36ad6c4adbe2121272bd0c4d239bb153ab/activerecord/test/cases/test_case.rb#L232